x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java

```diff
@@ -97,4 +97,35 @@
         });
     }
 
+    @SuppressWarnings("unchecked")
+    static void updateDocument(Map<String, Object> document, String fieldName, Object value) {
+        String[] fieldTokens = fieldName.split("\\.");
+        Map<String, Object> internalMap = document;
+        for (int i = 0; i < fieldTokens.length; i++) {
+            String token = fieldTokens[i];
+            if (i == fieldTokens.length - 1) {
+                if (internalMap.containsKey(token)) {
+                    logger.error("duplicate key value pairs key {} old value {} duplicate value {}",
```

Comment:
I think we should throw it's better to fail the data frame than creating a broken one. Incomplete/wrong data is really bad.

Reply:
@hendrikmuhs if that is the case here, then we should probably do that when we encounter an unsupported aggregation no?

Reply:
@benwtrent You are right! This is probably some leftover, now that we have validation and error handling in place. If you could fix those, that would be great, I am fine if its part of this PR.